### PR TITLE
Bwutils tweaks

### PR DESCRIPTION
**Pull request by dgdekoning**
Add numpy-style docstrings to some of the classes in the `bwutils` directory  which hopefully clarify the functionality.

In addition, create new methods in the MetaDataStore class to encapsulate repeated code.

Finally, altered the `inventory_df` method, which used the reserved `type` keyword to use `inventory_type` instead. (which lead to small edits in code using that function).

### CODE REVIEW

File: activity_browser/app/ui/tables/inventory.py [hunk 1]
```diff
@@ -124,8 +124,8 @@ def __init__(self):
         self.database_name = None
         self.dataframe = pd.DataFrame()
         self.technosphere = True
-        self.act_fields = lambda: [f for f in ['reference product', 'name', 'location', 'unit'] if f in AB_metadata.dataframe.columns]
-        self.ef_fields = lambda: [f for f in ['name', 'categories', 'type', 'unit'] if f in AB_metadata.dataframe.columns]
+        self.act_fields = lambda: AB_metadata.get_existing_fields(['reference product', 'name', 'location', 'unit'])
```
Comment (dgdekoning) [inline activity_browser/app/ui/tables/inventory.py]: Does this still need to be a lambda? Check and update if not.

File: activity_browser/app/bwutils/metadata.py [hunk 2]
```diff
@@ -102,5 +128,28 @@ def reset_metadata(self):
         self.dataframe = pd.DataFrame()
         self.databases = set()
 
+    def get_existing_fields(self, field_list):
+        """Return a list of fieldnames that exist in the current dataframe.
+        """
+        return [fn for fn in field_list if fn in self.dataframe.columns]
+
+    def get_metadata(self, field_keys_list, columns):
```
Comment (dgdekoning) [inline activity_browser/app/bwutils/metadata.py]: Clarify 'field_keys_list': Just make this 'keys' or 'metadata_keys'.

File: activity_browser/app/ui/tables/inventory.py (hunk 1 above)
Comment (dgdekoning) [reply activity_browser/app/ui/tables/inventory.py]: Yes, lambda needs to stay, act_fields and ef_fields are called as functions in the sync function further down.

File: activity_browser/app/bwutils/multilca.py [hunk 3]
```diff
@@ -9,18 +9,37 @@
...
+    CONTRIBUTION ANALYSIS.
 
-    Initialization creates `self.lca_scores`, which is a NumPy array of LCA scores, with rows of functional units and columns of LCIA methods. Ordering is the same as in the `calculation_setup`.
+    Parameters
+    ----------
+    cs_name : str
+        Name of the calculation setup
 
-    Class adapted from bw2calc.multi_lca.MultiLCA to include also CONTRIBUTION ANALYSIS.
+    Attributes
```
Comment (bsteubing) [inline activity_browser/app/bwutils/multilca.py]: this is still quite incomplete. Best introduce also the different vectors and matrices that are used for storing the LCA results (e.g. process contributions); and also the indexing dictionaries

File: activity_browser/app/bwutils/multilca.py [hunk 4]
```diff
@@ -59,52 +73,61 @@ def __init__(self, cs_name):
...
         self.scaling_factors = dict()
-        self.technosphere_flows = dict()  # Technosphere product flows for a given functional unit
 
-        self.inventory = dict()  # Life cycle inventory (biosphere flows) by functional unit
-        self.inventories = dict()  # Inventory (biosphere flows) by activity (e.g. 2000x15000) and functional unit.
+        # Technosphere product flows for a given functional unit
+        self.technosphere_flows = dict()
+        # Life cycle inventory (biosphere flows) by functional unit
+        self.inventory = dict()
+        # Inventory (biosphere flows) by activity (e.g. 2000x15000) and functional unit.
```
Comment (bsteubing) [inline activity_browser/app/bwutils/multilca.py]: for a specific functional unit and LCIA method

File: activity_browser/app/bwutils/multilca.py [hunk 5]
```diff
@@ -141,25 +168,85 @@ def get_all_metadata(self):
 
 
 class Contributions(object):
-    """Contribution Analysis built on top of the Multi-LCA class."""
+    """Contribution Analysis built on top of the Multi-LCA class.
+
+    This class requires instantiated MLCA and MetaDataFrame objects.
```
Comment (bsteubing) [inline activity_browser/app/bwutils/multilca.py]: what is the right name now MetaDataFrame or MetaDataStore ? I have seen both now :-)

Comment (dgdekoning) [reply activity_browser/app/bwutils/multilca.py]: Whoops, you're right, have been using them interchangeably, will correct to MetaDataStore.

File: activity_browser/app/bwutils/multilca.py (hunk 3 above)
Comment (dgdekoning) [reply activity_browser/app/bwutils/multilca.py]: Have just added all of the attributes of the class. It kind of feels like the comments in the __init__ method are superfluous now.

File: activity_browser/app/bwutils/multilca.py [hunk 6]
```diff
@@ -9,18 +9,37 @@
...
+    This class does not subclass the `LCA` class, and performs all
+    calculations upon instantiation.
+
+    Initialization creates `self.lca_scores`, which is a NumPy array
+    of LCA scores, with rows of functional units and columns of LCIA methods.
+    Ordering is the same as in the `calculation_setup`.
 
-    This class does not subclass the `LCA` class, and performs all calculations upon instantiation.
+    Class adapted from bw2calc.multi_lca.MultiLCA to include also
+    CONTRIBUTION ANALYSIS.
```
Comment (bsteubing) [inline activity_browser/app/bwutils/multilca.py]: perhaps write something like this "process and elementary flow contributions and other LCA results (see attributes below)"
came across this as we also implement our own contribution analysis class...

Comment (dgdekoning) [reply activity_browser/app/bwutils/multilca.py]: How about:

> This class is adapted from `bw2calc.multi_lca.MultiLCA` and includes a
    number of additional attributes required to perform process- and
    elementary flow contribution analysis (see class `Contributions` below).

**Pull request merged**